avro/swagger/src/main/java/com/cerner/beadledom/avro/SwaggerAvroModelConverter.java

```diff
@@ -119,49 +115,33 @@ protected String getName(Schema schema) {
    * determined merely by looking at the schema. It may also be used recursively to build collection
    * and union types.
    *
-   * @return the parsed property, or null if it cannot/should not be represented in the Swagger
-   *     model
+   * @return the parsed property, or null if it cannot/should not be represented in the Swagger model
    */
   @Nullable
-  protected ModelProperty parseSchema(Schema schema) {
+  protected Property parseSchema(Schema schema) {
     switch (schema.getType()) {
       case RECORD:
-        return simpleProperty(getName(schema), schema.getFullName());
+        return PropertyBuilder.build(getName(schema), schema.getFullName(), Collections.emptyMap());
       case ENUM:
         // TODO may wish to include the enum's documentation in the field documentation, since it
         // won't appear anywhere else
-        return new ModelProperty(
-            "string",
-            "string",
-            0,
-            true,
-            Option.<String>empty(),
-            new AllowableListValues(
-                JavaConversions.asScalaBuffer(schema.getEnumSymbols()).toList(),
-                "LIST"),
-            Option.<ModelRef>empty()
-        );
+        StringProperty property = new StringProperty();
+        property.required(true);
+        property.setEnum(schema.getEnumSymbols());
+
+        return property;
       case ARRAY:
-        ModelProperty elementsProperty = parseSchema(schema.getElementType());
+        Property elementsProperty = parseSchema(schema.getElementType());
         if (elementsProperty == null) {
           return null;
         }
 
-        if (SwaggerSpec.containerTypes().contains(elementsProperty.type())) {
```

Comment:
I take it this is now always allowed?

Reply:
No, that is to prevent multi-dimensional arrays. There is no longer a `SwaggerSpec` class so looking if there is a replacement.

Reply:
maybe 0970fb8d

Reply:
fe0ff756